### ISSUE TITLE
Show Number Incorrect When Use "emojione-android.ttf" 

### ISSUE THREAD

**Issue by saeedrooyekhat** [Ranks/emojione#385]
when using from emojione-android.ttf in app , when writing number , show number like emoji !!!

i write "1" but show this number like emoji . how fix it?

**Comment by miguelsousa** [comment]
Are you using the latest version of the font?
Seems like a duplicate of #305.

**Comment by Naudiz** [comment]
Wait, have the keycaps been COMPLETELY removed from the ttf??? This would be pretty bad in my opinion.
On my keyboard they seem to be totallyabsent since updating. And I have also seen other people reporting weird behaviour like this:
![attachment php](https://cloud.githubusercontent.com/assets/16839938/19973992/3c9c6848-a1e7-11e6-9bc0-c6f63e1e2668.png)

IMHO they need to be definitely there, or else the font is incomplete. To ensure that not normal numbers are displayed as emoji, that's what usually ZWJs and fe0f characters are there for.

**Comment by Naudiz** [comment]
@miguelsousa I looked into the latest version of your font, and in my opinion you removed too much. 

What you should have removed and what caused the bug #305 where the characters:
u0023
u002A
u0030
u0031
u0032
u0033
u0034
u0035
u0036
u0037
u0038
u0039

Instead, you removed all of the above, but also:
u0023_20E3
u002A_20E3
u0030_20E3
u0031_20E3
u0032_20E3
u0033_20E3
u0034_20E3
u0035_20E3
u0036_20E3
u0037_20E3
u0038_20E3
u0039_20E3

All those with 20E3 at the end NEED to be in any complete emoji font though and they also don't cause problems like #305 
Please restore those to the next update. The current version of the font is really broken.

**Comment by miguelsousa** [comment]
@Naudiz it's not possible to implement the *_20E3 glyphs if the font cannot contain the first characters of the sequence (i.e. your first list).

I agree that the font is a lot more complete (and better) like it was before, but other people were having problems. If you can provide a solution to those people, I'll gladly restore the keycaps.

**Comment by Naudiz** [comment]
@miguelsousa Okay, I understand.

I investigated a little bit further and found out that even the stock Android emoji font NotoColorEmoji.ttf contains graphics for 0023, 002A, 0030, etc. (the bare figures without 20E3), so I would search the reason for bug #305 at the respective user's setup first.

Have those users maybe edited /system/etc/fonts.xml? It defines the order in which fonts are used. Usually, when I type the number 1, Android will pull it from one of the non-emoji fonts. Only when font.xml is modified in a way that the priority of the Emoji TTF is too high, the system will look for the figure 1 there first, thus displaying the Emoji.

In short terms: Generally a character will only be displayed when no other font higher in the list of fonts.xml does contain this character.

The fact that even the stock-TTF has those figures and doesn't display them usually, and also that many people don't have the same problem, makes me believe that something in the user's setup is wrong rather than in your font.

**Comment by Naudiz** [comment]
Also, the users who reported the bugs #305 and #385 both stated that they used the font in an app.

App-developers obviously have to make sure that they have a proper fallback-solution implemented just like Android does, where figures 1-9, 0, #, * are pulled from a proper non-emoji font and only the remaining figures from EmojiOne.ttf.

**Comment by miguelsousa** [comment]
@Naudiz I agree with you on all counts.
@caseyahenson what should be done here? Bring back the keycaps and tell people with problems that they need to fix them on their end?

**Comment by caseyahenson** [comment]
This has been a pretty productive discussion, thanks @Naudiz for the info on fonts.xml ordering. @miguelsousa since it sounds like we now have a solution to recommend to developers who were experiencing the issue, I think it would be great to go back to the previous font file.

**Comment by miguelsousa** [comment]
@caseyahenson will do. Thanks for the quick reply.

**Comment by miguelsousa** [comment]
@Naudiz can you please open an issue?

**Comment by Naudiz** [comment]
@miguelsousa Added issue #388

**Comment by caseyahenson** [comment]
I'm adding this issue into the font README since it's still relevant, but resolved.

**Comment by miquelbeltran** [comment]
> App-developers obviously have to make sure that they have a proper fallback-solution implemented just like Android does, where figures 1-9, 0, #, * are pulled from a proper non-emoji font and only the remaining figures from EmojiOne.ttf.

Sorry to ask. What's the recommended way to do that? For example, using a TypefaceSpan for the emoji characters? or is there a better way? Thanks a lot for your work!

**Comment by foxyoverdrive** [comment]
I have the same question as the @miquelbeltran

**Comment by mcmar** [comment]
Wait, so I'm not 100% sure what's going on here. I can't just add this font to a `<Text>` element in react-native with this font in my styles?
I now have to:
1: Create a custom component
2: Parse the string for emoji characters
3: Replace any strings of emojis with a new `<Text>` element containing this font
4. Sequence my new series of strings and text elements inside the component
5: If there were already any `<Text>` elements inside, then recursively call my same component to re-render them using the same emoji algorithm
6: Re-sequence that entire list of strings and `<Text>` elements back together in the correct order.

So basically, you're saying that I need to re-create this library for react-native:
https://github.com/pladaria/react-emojione#readme
That library already exists for react on the web and it replaces text spans with the image nodes instead of using fonts. But the whole reason I came here was because I want to use emojione in my react-native application and there's no library that can help me do that. I really think we should go back to the way this was before because now this font is completely useless for application developers. I can't have all my numbers being replaced with the blue circle numbers. And I can't be expected to re-create a large and complex library just to integrate this into an application. Can you please reverse course and remove the numbers from this font?

**Comment by terijyu** [comment]
@miquelbeltran @caseyahenson +1, experiencing this on react-native. Please help! 🙏

**Comment by polyn0m** [comment]
@mcmar @caseyahenson  +1, font must include normal numbers, normal space. And include emoji keycaps by ZWJ or fe0f sequences. Now it font difficult and expensive to use, because it's need many excess actions for display that emoji.

**Comment by szarox7** [comment]
> @mcmar @caseyahenson +1, font must include normal numbers, normal space. And include emoji keycaps by ZWJ or fe0f sequences. Now it font difficult and expensive to use in Android applications, because it's need many excess actions for display that emoji.

Hey man, after 1 years, do you have sucess to create this police or anyone succes to create this? because I don't understand the all instructions but I'm interested to this project to use on my phone 😊
Thanks to reply
